examples/fungible-token/src/state.rs

```diff
@@ -0,0 +1,299 @@
+use fvm_actors_runtime::Map;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::Cbor;
+use fvm_ipld_hamt::BytesKey;
+use fvm_primitives::{TCid, THamt};
+use fvm_shared::{address::Address, bigint::Zero, econ::TokenAmount, ActorID, HAMT_BIT_WIDTH};
+use integer_encoding::VarInt;
+use ipc_atomic_execution::{AtomicExecID, AtomicExecRegistry, AtomicInputID, AtomicInputState};
+use ipc_gateway::IPCAddress;
+use serde::{Deserialize, Serialize};
+use serde_tuple::{Deserialize_tuple, Serialize_tuple};
+use std::collections::HashMap;
+
+pub type AccountState = AtomicInputState<TokenAmount>;
+
+#[derive(Serialize, Deserialize)]
+pub struct State {
+    ipc_gateway: Address,
+    ipc_address: IPCAddress,
+    name: String,
+    symbol: String,
+    total: TokenAmount,
+    balances: TCid<THamt<ActorID, AccountState>>,
+    atomic_registry: AtomicExecRegistry,
+}
+impl Cbor for State {}
+
+impl State {
+    pub fn new(
+        bs: &impl Blockstore,
+        ipc_gateway: Address,
+        ipc_address: IPCAddress,
+        name: String,
+        symbol: String,
+        balances: impl IntoIterator<Item = (ActorID, TokenAmount)>,
```

Comment:
I see! Are providing a balance table in the constructor so you can set a token pre-allocation to avoid having to implement a minting function and a circulating supply? That makes sense.

Reply:
Exactly. We could also supply credentials of the owner, who is allowed to mint tokens, but that would be more complicated than necessary for this example.

Reply:
This approach works, thank you for the clarification.